### PR TITLE
Add canal version bump test

### PR DESCRIPTION
**Pull request by fmoral2**
#### Proposed Changes ####

- add: canal version bump test 

#### Types of Changes ####

- Addition of tests and minor corrections

#### Verification ####

- Tests ran 4 times in rke2 only

#### Testing ####

- Ran each test with all configurations 4 times- OK
- Code lint: OK
- Make file | Docker Run | Jenkins updated: NOK  -> see here https://github.com/rancher/qa-tasks/issues/854

### CODE REVIEW

File: workloads/amd64/upgrade-plan.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
+apiVersion: upgrade.cattle.io/v1
+kind: Plan
+metadata:
+  name: rke2-server-cp
```
Comment (mdrahman-suse) [inline workloads/amd64/upgrade-plan.yaml:4]: should this be changed to {PRODUCT}-server-cp?

File: workloads/amd64/upgrade-plan.yaml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  version: $UPGRADEVERSION
+  nodeSelector:
+    matchExpressions:
+      - {key: node-role.kubernetes.io/control-plane, operator: In, values: ["true"]}
+  tolerations:
+    - operator: Exists
+  serviceAccountName: system-upgrade
+  cordon: true
+  upgrade:
+    image: rancher/rke2-upgrade
```
Comment (mdrahman-suse) [inline workloads/amd64/upgrade-plan.yaml:19]: {PRODUCT}-upgrade

File: workloads/amd64/upgrade-plan.yaml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+---
+apiVersion: upgrade.cattle.io/v1
+kind: Plan
+metadata:
+  name: rke2-server-etcd
```
Comment (mdrahman-suse) [inline workloads/amd64/upgrade-plan.yaml:24]: {PRODUCT}-server-etcd

File: workloads/amd64/upgrade-plan.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+          operator: In
+          values: [ "true" ]
+        - key: node-role.kubernetes.io/control-plane
+          operator: NotIn
+          values: [ "true" ]
+  tolerations:
+    - operator: Exists
+  serviceAccountName: system-upgrade
+  prepare:
+    image: rancher/rke2-upgrade
```
Comment (mdrahman-suse) [inline workloads/amd64/upgrade-plan.yaml:45]: {PRODUCT}-upgrade

File: workloads/amd64/upgrade-plan.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    args: ["prepare", "rke2-server-cp"]
```
Comment (mdrahman-suse) [inline workloads/amd64/upgrade-plan.yaml:46]: {PRODUCT}-server-cp

File: workloads/amd64/upgrade-plan.yaml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  cordon: true
+  drain:
+    force: true
+  upgrade:
+    image: rancher/rke2-upgrade
```
Comment (mdrahman-suse) [inline workloads/amd64/upgrade-plan.yaml:51]: {PRODUCT}-upgrade

File: workloads/amd64/upgrade-plan.yaml [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+---
+apiVersion: upgrade.cattle.io/v1
+kind: Plan
+metadata:
+  name: rke2-agent
```
Comment (mdrahman-suse) [inline workloads/amd64/upgrade-plan.yaml:56]: {PRODUCT}-agent

File: workloads/amd64/upgrade-plan.yaml [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+spec:
+  concurrency: 2
+  version: $UPGRADEVERSION
+  nodeSelector:
+    matchExpressions:
+      - {key: node-role.kubernetes.io/etcd, operator: NotIn, values: ["true"]}
+      - {key: node-role.kubernetes.io/control-plane, operator: NotIn, values: ["true"]}
+  serviceAccountName: system-upgrade
+  prepare:
+    image: rancher/rke2-upgrade
```
Comment (mdrahman-suse) [inline workloads/amd64/upgrade-plan.yaml:69]: {PRODUCT}-upgrade

File: workloads/amd64/upgrade-plan.yaml [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    args: ["prepare", "rke2-server-etcd"]
```
Comment (mdrahman-suse) [inline workloads/amd64/upgrade-plan.yaml:70]: {PRODUCT}-server-etcd

File: workloads/amd64/upgrade-plan.yaml [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+  drain:
+    force: true
+  upgrade:
+    image: rancher/rke2-upgrade
```
Comment (mdrahman-suse) [inline workloads/amd64/upgrade-plan.yaml:74]: {PRODUCT}-upgrade

File: workloads/amd64/upgrade-plan.yaml (hunk 2 above)
Comment (fmoral2) [reply workloads/amd64/upgrade-plan.yaml:19]: same

File: .gitignore [hunk 11]
```diff
@@ -4,7 +4,7 @@
 *.modules/.terraform/
 *.terraform*
 *.tfstate*
-*plan.yaml
+workloads/amd64/plan.yaml
```
Comment (mdrahman-suse) [inline .gitignore:7]: should we be ignoring the plan.yaml?

Comment (fmoral2) [reply .gitignore:7]: yes...since its created in each upgrade running a new one.

**Review by VestigeJ — APPROVED** [review]
LGTM - thank you Fran / MD

**Review by mdrahman-suse — APPROVED** [review]
LGTM

**Pull request merged**